app/lib/foreman_deployments/registry.rb

```diff
@@ -0,0 +1,67 @@
+module ForemanDeployments
+  class Registry
+    class TypeException < ::Foreman::Exception; end
+
+    ALLOWED_TYPES = {
+      'task' => 'ForemanDeployments::Tasks::BaseDefinition',
+      'input' => 'ForemanDeployments::Inputs::BaseInputDefinition'
+    }
+
+    def initialize
+      clear!
+    end
+
+    def clear!
+      @available = Hash[ALLOWED_TYPES.keys.map { |k| [k, {}] }]
+    end
+
+    def available(type)
+      @available[type].clone
+    end
+
+    def register(type, registered_class)
```

Comment:
Please no meta-definitions when it's not necessary. You save a few lines of code but loose a lot of readability.